<a href="https://colab.research.google.com/github/kartik-555/data-analytics/blob/main/data_analysis.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
from google.colab import drive
drive.mount('/content/drive')

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


Importing Libraries

In [ ]:
import pandas as pd


Importing file

In [ ]:
data=pd.read_excel('/content/Applications_for_Machine_Learning_internship_edited.xlsx')

In this cell
1. add index to empty cells of column Name
2.deleting rows with 0 value in any of four python, machine, NLP, Deep Learning
3.find the average and convert to percentage

In [ ]:
data['Name']=range(1,len(data)+1)
bcolumns=['Python (out of 3)','Machine Learning (out of 3)','Natural Language Processing (NLP) (out of 3)','Deep Learning (out of 3)']
pcolumns=['Performance_UG','Performance_12','Performance_10']
data=data.dropna(subset=pcolumns)
data = data[~(data[bcolumns] == 0).any(axis=1)]
data['Average']=data[['Python (out of 3)','Machine Learning (out of 3)','Natural Language Processing (NLP) (out of 3)','Deep Learning (out of 3)']].mean(axis=1)
data['Average'] = (data['Average'] / 3) * 100

In this cell
1. Deleting values rows with null values in performance_UG,Performance_12 Performance_10
2.Normalizing values values that are out of 10 converting them to out of 100
3. Removing values after '/' because there are many errors in them
4. Then finding average of performances in three feilds (performance pg is not that much necessary and does not contain that much weightage)
5. Finding average of performance and skills
6. Putting them in decending order

  **TOP MOST PERSON IS IDEAL FOR INTERNSHIP**

In [ ]:
def transform_values(column):
    transformed_column = column.copy()
    for i, value in column.iteritems():
        if '/' in value:
            parts = value.split('/')
            transformed_column[i] = parts[0]
        elif float(value) < 10:
            transformed_column[i] = str(float(value) * 10)
    return transformed_column

columns_to_transform = ['Performance_UG', 'Performance_12', 'Performance_10']
data[columns_to_transform] = data[columns_to_transform].apply(transform_values)


# Convert values less than 10 to float and multiply by 10 for 'Performance_UG'
data.loc[data['Performance_UG'].astype(float) < 10, 'Performance_UG'] = (data['Performance_UG'].astype(float) * 10).astype(str)

# Convert values less than 10 to float and multiply by 10 for 'Performance_12'
data.loc[data['Performance_12'].astype(float) < 10, 'Performance_12'] = (data['Performance_12'].astype(float) * 10).astype(str)

# Convert values less than 10 to float and multiply by 10 for 'Performance_10'
data.loc[data['Performance_10'].astype(float) < 10, 'Performance_10'] = (data['Performance_10'].astype(float) * 10).astype(str)

# Compute the average of the three columns and store it in a new column 'Average'
data['perf_Average'] = data[['Performance_UG', 'Performance_12', 'Performance_10']].astype(float).mean(axis=1)


data['order'] = (data['perf_Average'] + data['Average']) / 2
data_sorted = data.sort_values(by='order', ascending=False)

data_sorted

<ipython-input-44-c9611bfa1724>:3: FutureWarning: iteritems is deprecated and will be removed in a future version. Use .items instead.
  for i, value in column.iteritems():


Name  Python (out of 3)  Machine Learning (out of 3)  \
862    863                  3                            3   
767    768                  3                            3   
1073  1074                  3                            3   
246    247                  3                            3   
404    405                  3                            3   
...    ...                ...                          ...   
656    657                  3                            2   
29      30                  2                            2   
748    749                  2                            1   
662    663                  2                            2   
981    982                  1                            1   

      Natural Language Processing (NLP) (out of 3)  Deep Learning (out of 3)  \
862                                              3                         3   
767                                              3                         3   
1073                                             3                         3   
246                                              3                         3   
404                                              2                         2   
...                                            ...                       ...   
656                                              2                         2   
29                                               1                         1   
748                                              1                         2   
662                                              1                         1   
981                                              1                         1   

                                           Other skills  \
862   C++ Programming, Computer Vision, Data Analyti...   
767   Computer Vision, Data Analytics, Data Science,...   
1073  Data Analytics, Data Science, Deep Learning, M...   
246   Deep Learning, Flask, Hadoop, JavaScript, Mach...   
404   Data Analytics, English Proficiency (Spoken), ...   
...                                                 ...   
656   Data Science, MS-Excel, Python, R Programming,...   
29    Cloud Computing, Amazon Web Services (AWS), C ...   
748   HTML, Bootstrap, C Programming, C++ Programmin...   
662   Arduino, CSS, Data Science, Data Structures, E...   
981   English Proficiency (Spoken), MS-Word, C Progr...   

     Are you available for 3 months, starting immediately, for a full-time work from home internship?   \
862                                No, From 1st august.                                                  
767   Yes, I am available for 3 months starting imme...                                                  
1073  Yes, I am available for 3 months starting imme...                                                  
246   Yes, I am available for 3 months starting imme...                                                  
404   Yes, I am available for 3 months starting imme...                                                  
...                                                 ...                                                  
656   Yes, I am available for 3 months starting imme...                                                  
29    Yes, I am available for 3 months starting imme...                                                  
748   Yes, I am available for 3 months starting imme...                                                  
662   Yes, I am available for 3 months starting imme...                                                  
981   Yes, I am available for 3 months starting imme...                                                  

                                     Degree  \
862                                  B.Tech   
767              Integrated B.Tech & M.Tech   
1073                                 B.Tech   
246                Master of Science (M.Sc)   
404                                  B.Tech   
...                                 

**Results**

Person on top at Name= 862 is ideal

after that Name= 768 is ideal